app.js

```diff
@@ -1,10 +1,53 @@
 const WebSocket = require('ws');
 require('dotenv').load();
-var exp = require('./index.js');
 
+// when socket opens, notify channel
+function onOpen (soc) {
+    console.log('socket opened');
+    var msg = {
+        "id": 1,
+        "type": "message",
+        "channel": "C1BBWJ7PF",
+        "text": "WhoopBot connected to WebSocket"
+    };
+    
+    if (soc.readyState == 1) {
+        soc.send(JSON.stringify(msg));
+    };
+};
 
+// log incoming messages
+function onMessage (event) {
+    var ev = JSON.parse(event);
+    var text = ev.text;
+    console.log(event);
+};
+
+
+// when HTTPS request finished, initialize WebSocket and handle events
+function finishedRequest(data) {
+    //console.log('running fin');
+    //console.log(data);
+    //console.log('ran fin');
+    var url = JSON.parse(data).url;
+    //var selfId = JSON.parse(data).self.id
+    // console.log(url);
+
+    var socket = new WebSocket(url);
+
+    // handle socket opening
+    socket.on('open', function() {
+        onOpen(socket);
+    });
+
+    // handle incoming messages
+    socket.on('message', function(event) {
+        onMessage(event);
+    });
+
+};
 
 module.exports = {
-    onOpen: onOpen,
-    onMessage: onMessage
-};
\ No newline at end of file
+    fin: finishedRequest
```

Comment:
I agree with everything Nate said.

I would also go further and say that it's even more important for module export names to be fully self-explanatory than regular variables -- so if it did make sense for some reason to use both of these names to refer to the same function, then you'd be using the short name in this file and exporting the long name for use elsewhere -- not vice versa.